### PR TITLE
Update: add `tag` output to goreleaser-action

### PR DESCRIPTION
**Pull request by till**

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -48,6 +48,14 @@ async function run(): Promise<void> {
     if (!('GORELEASER_CURRENT_TAG' in process.env)) {
       process.env.GORELEASER_CURRENT_TAG = tag;
     }
+
+    if (isTagDirty) {
+      core.setOutput("tag", `${tag}-next`);
+    } else {
+      core.setOutput("tag", tag);
+    }
```
Comment (crazy-max) [inline src/main.ts:56]: I don't think we should have `-next` hardcoded like this. It should reflect the `.goreleaser.yml` provided.

I wonder if somehow GoReleaser should produce a metadata file (e.g., `metadata.json`) that contains the list of [known/generated templates vars](https://goreleaser.com/customization/templates/#name-templates) like `.Version`, etc... so we could use it as a `metadata` output in GitHub Actions:

```yaml
      -
        name: Run GoReleaser
        id: goreleaser
        uses: goreleaser/goreleaser-action@v2
        with:
          version: latest
          args: release --rm-dist
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
      -
        run: |
          echo "${{ fromJSON(steps.goreleaser.outputs.metadata).version }}"
```

Thoughts @caarlos0?

Comment (till) [reply src/main.ts:56]: That would work as well. Just looking to deploy a fluent version.

Comment (till) [reply src/main.ts:56]: @crazy-max Btw can you point out how `.goreleaser.yml` can influence the name? Wasn't aware of that.

Comment (caarlos0) [reply src/main.ts:56]: I thinks it's a good idea @crazy-max , we already do it for artifacts...

Comment (caarlos0) [reply src/main.ts:56]: @till the snapshot name template config 


And also the nightly on goreleaser pro

Comment (till) [reply src/main.ts:56]: @caarlos0 Would you be open if I adjust that then? I don't want to parse another file in my workflow. An output would be preferred. :)

Comment (caarlos0) [reply src/main.ts:56]: yeah, we need to add this feature to goreleaser first (e.g. a `dist/context.json`) or something like that, the IMHO we can parse it here and output the values...

things I think would be useful as outputs:

- current and previous tag names and versions
- project name
- what else?

artifacts can be get from `dist/artifacts.json`

Comment (till) [reply src/main.ts:56]: Maybe isDirty, or isSnapshot? Can't think of much right now.

Comment (caarlos0) [reply src/main.ts:56]: yeah, sound good, will open an issue on goreleaser

Comment (caarlos0) [reply src/main.ts:56]: https://github.com/goreleaser/goreleaser/issues/2835

Comment (crazy-max) [reply src/main.ts:56]: > and output the values...

Yeah we just need one output named `metadata` or `context` that will be the JSON representation so we can use `fromJSON()` in the underlying steps.

We do this in the [metadata-action](https://github.com/docker/metadata-action#json-output-object):

https://github.com/docker/metadata-action/blob/2cad85de2462ab92a22ff973a347fb25c91d6082/src/main.ts#L62-L67

image

> what else?

The more the better :pray:

**Pull request closed without merging**